pandas/core/common.py

```diff
@@ -1482,10 +1482,11 @@ def _get_dtype_from_object(dtype):
 
         try:
             return _get_dtype_from_object(getattr(np,dtype))
```

Comment:
hmm, I think this should be changed to:

``np.dtype(dtype)`` as that wontt have the issue, and catch ``TypeError`` and not ``AttributeError``

and is more correct